src/backend/columnar/columnar_tableam.c

```diff
@@ -1896,6 +1911,61 @@ ColumnarTriggerCreateHook(Oid tgid)
 }
 
 
+/*
+ * ColumnarTableCreateHook is called by ObjectAccessHook to perform the
+ * operations that should be done after creating a columnar table.
+ */
+static void
+ColumnarTableCreateHook(Oid relationId)
+{
+	if (!IsColumnarTableAmTable(relationId))
+	{
+		return;
+	}
+
+	if (RelationUsesPgColumnCompression(relationId))
+	{
+		ereport(NOTICE, (errmsg(COLUMNAR_PG_COMPRESSION_NOTICE_MSG)));
+	}
+}
+
+
+/*
+ * RelationUsesPgColumnCompression returns true if given relation uses
+ * built-in column compression for any columns.
+ *
+ * Note that this is different than the column compression functionality
+ * provided by columnarAM.
+ */
+static bool
+RelationUsesPgColumnCompression(Oid relationId)
+{
+	bool relationUsesPgColumnCompression = false;
+
+#if PG_VERSION_NUM >= 140000
+	Relation relation = RelationIdGetRelation(relationId);
+	TupleDesc tupleDescriptor = RelationGetDescr(relation);
+	for (int columnIndex = 0; columnIndex < tupleDescriptor->natts; columnIndex++)
+	{
+		Form_pg_attribute attributeForm = TupleDescAttr(tupleDescriptor, columnIndex);
+		if (attributeForm->attisdropped)
+		{
+			continue;
+		}
```

Comment:
Could it ever be the case ? Maybe when we are inheriting a table that had some dropped columns.
Will look into once more.

Reply:
Yes, we might encounter with dropped columns when vacuuming a table, as in below:

```sql
create table tbl (int_col INT, comp_text_1 TEXT COMPRESSION pglz) using columnar;
alter table tbl drop column int_col;

vacuum full tbl ;
NOTICE:  since columnar tableAM already compresses columns in large chunks, using built-in postgres compression might not bring additional compression benefits, see "alter_columnar_table_set" function and "compression" & "compression_level" arguments to use the compression functionality provided by columnarAM
```

But now I realized that our `CreateTableHook` gets called for table re-writes as well and we would print that notice message in such cases as well.
The question is, are we okay with that ? I think this is not important at all.